internal/jobworker/grpc/server.go

```diff
@@ -0,0 +1,230 @@
+// Package grpc provides types that implement a jobworker grpc server.
+package grpc
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/tjper/teleport/internal/jobworker/cgroup"
+	"github.com/tjper/teleport/internal/jobworker/job"
+	"github.com/tjper/teleport/internal/jobworker/reexec"
+	"github.com/tjper/teleport/internal/log"
+	"github.com/tjper/teleport/internal/validator"
+	pb "github.com/tjper/teleport/proto/gen/go/jobworker/v1"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// logger is an object for logging package events to stdout.
+var logger = log.New(os.Stdout, "grpc")
+
+// NewJobWorker creates a JobWorker instance.
+func NewJobWorker(jobSvc *job.Service, userSvc IUserService) *JobWorker {
+	return &JobWorker{jobSvc: jobSvc, userSvc: userSvc}
+}
+
+var _ pb.JobWorkerServiceServer = (*JobWorker)(nil)
+
+// IUserService provides an API for interacting with jobworker users.
+type IUserService interface {
+	// User retrieves the user associated with the ctx. The ok return value
+	// should indicate if the user could be retrieved. The user return value
+	// should be the user's unique identifer.
+	User(ctx context.Context) (string, bool)
+}
+
+// Jobworker provides mechanisms for starting, stopping, fetching status, and
+// output streaming jobs.
+// Jobworker implements pb.JobWorkerServiceServer.
+type JobWorker struct {
+	jobSvc  *job.Service
+	userSvc IUserService
+}
+
+func (jw JobWorker) Start(ctx context.Context, req *pb.StartRequest) (*pb.StartResponse, error) {
+	user, ok := jw.userSvc.User(ctx)
+	if !ok {
+		return nil, status.Error(codes.Unauthenticated, "unauthenticated")
+	}
+
+	valid := validator.New()
+	valid.AssertFunc(func() bool { return req.Command != nil }, "command empty")
+	valid.AssertFunc(func() bool { return req.Command.Name != "" }, "command name empty")
+	valid.AssertFunc(func() bool { return req.Limits != nil }, "limits empty")
```

Comment:
I would ditch the entire validator package for a switch:

```suggestion
	switch {
	case req.Command == nil:
		return nil, status.Errorf(...)
	case req.Command.Name == "":
		return nil, status.Errorf(...)
	case req.Limits == nil:
		return nil, status.Errorf(...)
	}
```

As a small benefit we even get conditions that are not inverted.